File: Easy/tooMany/tooMany.java
//run as java -ea tooMany
import java.util.Scanner;

public class tooMany{
	public static void main(String[] args){
		int T;
		long N,M,K,rem;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=100000) : "T must be 1 <= T <= 10^5";

		while(T>0){
			N = scan.nextLong();
			M = scan.nextLong();
			K = scan.nextLong();

			assert(N>=2 && N<=1000000000) : "N must be 2 <= N <= 10^9";
			assert(M>=2 && M<=1000000000) : "M must be 2 <= M <= 10^9";
			assert(K>=2 && K<=1000000000) : "K must be 2 <= K <= 10^9";

			if(M>N){
				System.out.println(-1);
			}else{
				rem = N%M;

				if(rem%K==0){
					System.out.println(rem/K);
				}else{
					System.out.println(-1);
				}
			}

			T--;
		}
	}
}

File: Easy/tooMany/tooMany.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T;
	long long int N,M,K,rem;

	scanf("%d",&T);
	assert(T>=1 && T<=100000);

	while(T>0){
		scanf("%lld %lld %lld",&N,&M,&K);

		assert(N>=2 && N<=1000000000);
		assert(M>=2 && M<=1000000000);
		assert(K>=2 && K<=1000000000);

		if(M>N){
			printf("-1\n");
		}else{
			rem = N%M;

			if(rem%K==0){
				printf("%lld\n",rem/K);
			}else{
				printf("-1\n");
			}
		}

		T--;
	}
	
	return 0;
}